content/vault/v1.21.x/content/docs/deploy/run-as-zcx-cluster.mdx

```diff
@@ -0,0 +1,317 @@
+  }
+
+  # --- Raft Storage (Leader Node) ---
+  storage "raft" {
+    path    = "/vault/data"
+    node_id = "<LEADER_NODE_ID>"  # e.g., "node1"
+
+    # No retry_join block needed for leader
+  }
+
+  # --- Cluster Networking ---
+  api_addr     = "https://<LEADER_IP>:8200"
+  cluster_addr = "https://<LEADER_IP>:8201"
+  ```
+
+  Exposes Vault on port 8200 for API and UI traffic.
+
+   - **`tls_disable`** = 0 → TLS is enabled (mandatory for secure cluster communication).
+
+### Certificates:
+
+- **`vault.pem`** → TLS certificate
+
+- **`vault.key`** → Private key
+
+- **`ca.pem`** → CA used to validate mutual TLS between nodes
+
+This ensures encrypted traffic between Vault clients, HAProxy, and other nodes.
+
+Similarly create configuration for follower Vault nodes.
+
+```hcl
+ui = true
+disable_mlock = true
+
+# --- Listener Configuration ---
+listener "tcp" {
+  address            = "0.0.0.0:8200"
+  tls_disable        = 0
+  tls_cert_file      = "/vault/vault.pem"
+  tls_key_file       = "/vault/vault.key"
+  tls_client_ca_file = "/vault/ca.pem"
+}
+
+# --- Raft Storage (Follower Node) ---
+storage "raft" {
+  path    = "/vault/data"
+  node_id = "<FOLLOWER_NODE_ID>"   # e.g., "node2" or "node3"
+
+  retry_join {
+    leader_api_addr      = "https://<LEADER_IP>:8200"
+    leader_ca_cert_file  = "/vault/ca.pem"
+  }
+}
+
+# --- Cluster Networking ---
+api_addr     = "https://<FOLLOWER_IP>:8200"
+cluster_addr = "https://<FOLLOWER_IP>:8201"
+```
+
+## Step 5: Vault container deployment
+
+Deploy Vault container on each zCX node.
+
+1. Set your Vault Enterprise license (export before running).
+
+  ```shell-session
+  $ export VAULT_LICENSE="02MV4UU43BK5HGYYTOJZWFQMTMNNEWU33JJVVGC..." 
+  ```
+
+1. Deploying the Vault Enterprise container on zCX.
+
+  ```shell-session
+  $ docker run -d \
+      --name vault-zcx \
+      --cap-add IPC_LOCK \
+      -p 8200:8200 \
+      -p 8201:8201 \
+      -v vault-config:/vault/config.d \
+      -v vault-data:/vault/data \
+      -e VAULT_LICENSE="$VAULT_LICENSE" \
+      hashicorp/vault-enterprise:1.19.12-ent \
+      server -config=/vault/config.d/
+  ```
+
+1. Verify container status.
+
+  ```shell-session
+  $ docker ps | grep vault-zcx
+  ```
+
+ 1. Check the container log. 
+
+  ```shell-session 
+  $ docker logs vault-zcx
+  ```
+
+## Cluster initialization and unsealing
+
+<Note>
+
+Perform the initialization only on the leader node.
+
+</Note>
+
+1. Initialize the leader Vault.
+
+  ```shell-session
+  $ docker exec -it vault-zcx vault operator init \
+    -format=json > /secure/location/vault-init.json
+  ```
+
+  By default, the `operator init` command generates five unseal keys with threshold of three.
+
+1. Extract unseal keys and root token.
+
+  ```shell-session
+  $ cat /secure/location/vault-init.json
+  ```
+
+1. Unseal Vault on all nodes. 
+
+  Vault requires a certain threshold of shares to reconstruct the unseal key. Unseal Vault with 3 different unseal keys (repeat on each node).
```

Comment:
```suggestion
```
How-to guides should focus on explaining how to do something, not how the things works. I recommend adding an item to "Before you start" that notes you must be familiar with how unsealing works in Vault.